clientconn.go

```diff
@@ -62,6 +62,11 @@ var (
 	errBalancerClosed = errors.New("grpc: balancer is closed")
 	// minimum time to give a connection to complete
 	minConnectTimeout = 20 * time.Second
+	// We use an accessor so that minConnectTimeout can be
+	// atomically read and updated while testing.
+	getMinConnectTimeout = func() time.Duration {
+		return minConnectTimeout
```

Comment:
Could you make minConnectTimeout a const, and move the var to be in the test only?